packages/scripts/src/utils/relay.ts

```diff
@@ -0,0 +1,162 @@
+  to: string;
+  value: BigNumberish;
+  data: BytesLike;
+}> => {
+  if (!tx.data) throw new Error("Transaction has no data");
+  const relayGovernanceData = await governorRootTunnel.populateTransaction.relayGovernance(targetAddress, tx.data);
+  console.log("RelayGovernanceData", relayGovernanceData);
+  const relay = await governorRootTunnel.populateTransaction.relayGovernance(targetAddress, tx.data);
+  const relayMessage = relay.data;
+  if (!relayMessage) throw new Error("Relay message is empty");
+  return { to: governorRootTunnel.address, value: 0, data: relayMessage };
+};
+
+export const relayGovernanceHubMessages = async (
+  messages: GovernanceMessages,
+  governorHub: GovernorHub,
+  chainId: BigNumberish
+): Promise<
+  {
+    to: string;
+    value: BigNumberish;
+    data: BytesLike;
+  }[]
+> => {
+  const calls = messages.map((message) => {
+    if (!message.tx.data) throw new Error("Transaction has no data");
+    return { to: message.targetAddress, data: message.tx.data };
+  });
+  const relayGovernanceData = await governorHub.populateTransaction.relayGovernance(chainId, calls);
+  const relayMessage = relayGovernanceData.data;
+  if (!relayMessage) throw new Error("Relay message is empty");
+  return [{ to: governorHub.address, value: 0, data: relayMessage }];
+};
+
+export const relayGovernanceMessages = async (
+  messages: GovernanceMessages,
+  l1Governor: GovernorHub | GovernorRootTunnel,
+  chainId: number
+): Promise<
+  {
+    to: string;
+    value: BigNumberish;
+    data: BytesLike;
+  }[]
+> => {
+  // The l1 governor for polygon is the GovernorRootTunnel and the l1 governor for the rest of l2's is the GovernorHub
+  const isPolygon = chainId === 137;
+  console.log("isPolygon", isPolygon);
+  if (isPolygon) {
+    const relayedMessages = [];
+    for (const message of messages) {
+      relayedMessages.push(
+        await relayGovernanceRootTunnelMessage(message.targetAddress, message.tx, l1Governor as GovernorRootTunnel)
+      );
+    }
+    return relayedMessages;
+  }
+  return relayGovernanceHubMessages(messages, l1Governor as GovernorHub, chainId);
+};
+
+export const fundArbitrumParentMessengerForRelays = async (
+  arbitrumParentMessenger: ArbitrumParentMessenger,
+  from: Signer,
+  totalNumberOfTransactions: BigNumberish
+): Promise<void> => {
+  // Sending a xchain transaction to Arbitrum will fail unless Arbitrum messenger has enough ETH to pay for message:
+  const l1CallValue = await arbitrumParentMessenger.getL1CallValue();
+  console.log(
+    `Arbitrum xchain messages require that the Arbitrum_ParentMessenger has at least a ${hre.ethers.utils.formatEther(
+      l1CallValue.mul(totalNumberOfTransactions)
+    )} ETH balance.`
+  );
+
+  const apmBalance = await arbitrumParentMessenger.provider.getBalance(arbitrumParentMessenger.address);
+
+  if (apmBalance.lt(l1CallValue.mul(totalNumberOfTransactions))) {
+    const amoutToSend = l1CallValue.mul(totalNumberOfTransactions).sub(apmBalance);
```

Comment:
nit: I think it's a bit easier to read if you put cost into a separate variable.
```suggestion
  const cost = l1CallValue.mul(totalNumberOfTransactions);
  if (apmBalance.lt(cost)) {
    const amoutToSend = cost.sub(apmBalance);
```

Reply:
Agreed